packages/analytics-client-common/src/session.ts

```diff
@@ -0,0 +1,7 @@
+export const isNewSession = (sessionTimeout: number, lastEventTime?: number): boolean => {
+  const _currentTime = Date.now();
+  const _lastEventTime = lastEventTime || Date.now();
+  const timeSinceLastEvent = _currentTime - _lastEventTime;
+
+  return timeSinceLastEvent > sessionTimeout;
```

Comment:
if it is the first event (timeSinceLastEvent is 0). should we return true?

Reply:
while the first event of the session is being instrumented, it would probably be during a state where timeSinceLastEvent > sessionTimeout. so if timeSinceLastEvent === 0 or more generally timeSinceLastEvent < sessionTimeout, a valid session existed by the time this logic executed